src/java.desktop/share/classes/java/awt/Robot.java

```diff
@@ -773,4 +786,238 @@ public synchronized String toString() {
+     * @throws  IllegalThreadStateException if called on the AWT event dispatching thread
+     * @see     #mousePress(int)
+     * @see     #mouseRelease(int)
+     * @see     InputEvent#getMaskForButton(int)
+     * @see     Toolkit#areExtraMouseButtonsEnabled()
+     * @see     java.awt.event.MouseEvent
+     * @since   26
+     */
+    public void click(int buttons) {
+        mousePress(buttons);
+        waitForIdle(20);
+        mouseRelease(buttons);
+        waitForIdle(20);
+    }
+
+    /**
+     * A convenience method that clicks mouse button 1.
+     *
+     * @throws  IllegalThreadStateException if called on the AWT event dispatching thread
+     * @see     #click(int)
+     * @since   26
+     */
+    public void click() {
+        click(InputEvent.BUTTON1_DOWN_MASK);
+    }
+
+    /**
+     * A convenience method that calls {@code waitForIdle} then waits an additional specified
+     * {@code delayValue} time in milliseconds.
+     *
+     * @param   delayValue  Additional delay length in milliseconds to wait until thread
+     *                      sync been completed
+     * @throws  IllegalThreadStateException if called on the AWT event
+     *          dispatching thread
+     * @throws  IllegalArgumentException if {@code delayValue} is not between {@code 0}
+     *          and {@code 60,000} milliseconds inclusive
+     * @since   26
+     */
+    public synchronized void waitForIdle(int delayValue) {
+        waitForIdle();
+        delay(delayValue);
+    }
+
+    /**
+     * A convenience method that moves the mouse in multiple
+     * steps from its current location to the destination coordinates. Invokes
+     * {@link #mouseMove(int, int) mouseMove} with a {@link #getGlideStepLength() step-length}
+     * and {@link #getGlideStepDelay() step-delay}.
+     *
+     * @param   x   Destination point x coordinate
+     * @param   y   Destination point y coordinate
+     *
+     * @throws  IllegalThreadStateException if called on the AWT event dispatching
+     *          thread and {@code isAutoWaitForIdle} would return true
+     * @see     #getGlideStepLength
+     * @see     #getGlideStepDelay
+     * @see     #glide(int, int, int, int)
+     * @since   26
+     */
+    public void glide(int x, int y) {
+        Point p = MouseInfo.getPointerInfo().getLocation();
+        glide(p.x, p.y, x, y);
+    }
+
+    /**
+     * A convenience method that moves the mouse in multiple
+     * steps from the source coordinates to the destination coordinates. Invokes
+     * {@link #mouseMove(int, int) mouseMove} with a {@link #getGlideStepLength() step-length}
+     * and {@link #getGlideStepDelay() step-delay}.
+     *
+     * @param   srcX   Source point x coordinate
+     * @param   srcY   Source point y coordinate
+     * @param   destX  Destination point x coordinate
+     * @param   destY  Destination point y coordinate
+     *
+     * @throws  IllegalArgumentException if {@code stepLength} is greater than the distance
```

Comment:
In the CSR Joe asked if glide(int, int) needed to throw IAE and it made me look here and I don't actually see where this code would throw that. 
And the old extended robot never had such a feature, did it ?
Also if you think of glide as a sequence of steps, then even with the default of "2" if the distance to move is say "5" after 2 steps you are one pixel away and then I presume we just step one pixel for the final step.
So how is that different than STARTING one pixel away ?

Perhaps we can just dispense with the IAE and if dist < steplength then there's just one step.